dice.js

```diff
@@ -0,0 +1,294 @@
+    let chosenDiceLength = 0;
+    let unavailableDice = 0;
+    for (let i = 0; i < NUMBER_OF_DICE; i++) {
+      chosenDiceLength += diceCounts[i];
+      if (!gameState.dice[i].available) unavailableDice++;
+    }
+    if (chosenDiceLength === NUMBER_OF_DICE) {
+      scoringOptions.push({ roll: "No scoring dice", score: 500 });
+      removeAllDice(removeAllDice);
+    } else if (chosenDiceLength + unavailableDice === 6) {
+      scoringOptions.push({ roll: "Zilch!", score: 0 });
+    } else {
+      scoringOptions.push("Choose some dice to see your options");
+    }
+  }
+
+  return scoringOptions;
+};
+
+// Changes to the next player's turn
+const nextPlayerTurn = (gameState) => {
+  // If the score limit has not been reached, it becomes the next player's turn
+  gameState.accumulatedPoints = 0;
+  gameState.potentialRollScore = 0;
+  gameState.currentPlayer =
+    gameState.currentPlayer < gameState.numberOfPlayers
+      ? gameState.currentPlayer + 1
+      : 1;
+  gameState.dice = rollDice();
+};
+
+/*
+while(nextPlayer.score < gameState.scoreLimit) {
+  clearTurnState();
+  nextPlayerTurn();
+}
+
+*/
+
+// Returns whether any of the dice chosen will give points
+const validDiceChosen = (gameState) => {
+  for (let i = 0; i < NUMBER_OF_DICE; i++) {
+    if (gameState.dice[i].scored) {
+      return true;
+    }
+  }
+};
+
+// Sets the scored value for all dice used in the current combination false so they won't be scored again
+const removeScoredDice = (diceValue, iterations, chosenDice) => {
+  let currentIterations = 0;
+  for (let i = 0; i < NUMBER_OF_DICE; i++) {
+    if (
+      gameState.dice[i].value === diceValue &&
+      gameState.dice[i].available &&
+      chosenDice[i]
+    ) {
+      gameState.dice[i].scored = true;
+      currentIterations++;
+    }
+    if (currentIterations === iterations) break;
+  }
+};
+
+// Sets the scored value for all dice false, meaning they've all been scored (for combinations required all 6 dice)
+const removeAllDice = (gameState) => {
+  for (i = 0; i < NUMBER_OF_DICE; i++) {
+    gameState.dice[i].scored = true;
+  }
+};
+
+// Resets the scored value to false to allow for scoring
+// ** I don't understand why the dice scored attribute are being reset to false every time we try to get the scoring options
+const resetScoredDice = (gameState) => {
+  for (i = 0; i < NUMBER_OF_DICE; i++) {
+    gameState.dice[i].scored = false;
+  }
+};
+
+// Sets dice that have been previously used in a combo in the same turn to be unavailable
+const disableUsedDice = (selectedDice, gameState) => {
+  for (let i = 0; i < NUMBER_OF_DICE; i++) {
+    if (
+      gameState.dice[i].available &&
+      selectedDice[i] &&
+      gameState.dice[i].scored
+    )
+      gameState.dice[i].available = false;
+  }
+};
+
+// Enables the Roll Dice button when at least one valid dice has been chosen this roll.
+const enableRollDice = (gameState) => {
+  return validDiceChosen(gameState);
+};
```

Comment:
it doesn't seem like this function is doing anything that `validDiceChosen` isn't - candidate for removal?